src/database/DataTypes/Transaction.js

```diff
@@ -273,7 +274,10 @@ export class Transaction extends Realm.Object {
    */
   finalise(database) {
     if (this.isFinalised) throw new Error('Cannot finalise as transaction is already finalised');
+    // Should prune all invoice except internal supplier invoices
+    if (!this.isInternalSupplierInvoice) this.pruneRedundantBatchesAndItems(database);
```

Comment:
Fixes pruning on customer invoices.

Reply:
@andreievg FYI, customer invoices are 'confirmed' from the get go (hence adjusting stock before being finalised), so they never run confirm() code where the pruning had been moved to.

Reply:
We should also not prune batches for customer invoices that have linked requisition

Reply:
Actually rather then adding `isInternalSupplierInvoice`, we can use `isLinkedToRequisition`, no extra getter needed and covers both cases (supplier invoice and customer invoice -> is stock transfer)